### ISSUE TITLE
S3 stanza-create error with DELL ECS storage provider

### ISSUE THREAD

**Issue by kikijolicoeur** [pgbackrest/pgbackrest#877]
Hello. 

1. pgBackRest version:
2.19
2. PostgreSQL version:
10.6
3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
debian 9 (stretch)

4. Did you install pgBackRest from source or from a package?
installation from a package (repo http://apt.postgresql.org/pub/repos/apt/) : 2.19-1.pgdg90+1

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
[global]
process-max=2
repo1-type=s3
repo1-s3-bucket=srh
repo-path=/pitr/vvhnoupn7cak56zkqdvu
repo1-s3-endpoint=storage-eb4.cegedim.cloud
repo1-s3-key=****
repo1-s3-key-secret=******
repo1-s3-region=eu-west-3
repo1-retention-full=1
repo1-retention-diff=2
start-fast=y
stop-auto=y

[rebsrhpg001]
pg1-path=/var/lib/postgresql/data/10/main
pg1-socket-path=/var/run/postgresql

    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)

archive_mode = on
archive_command = 'pgbackrest --stanza=rebsrhpg001 archive-push %p'

    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

ERROR: [039]: S3 request failed with 400: Bad Request
       *** URI/Query ***:
       /?delimiter=0.000000&list-type=2&prefix=pitr0.000000vvhnoupn7cak56zkqdvu0.000000archive0.000000rebsrhpg0010.000000
       *** Request Headers ***:
       authorization: <redacted>
       content-length: 0
       host: srh.storage-eb4.cegedim.cloud
       x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
       x-amz-date: <redacted>
       *** Response Headers ***:
       connection: close
       content-type: application/xml
       date: Tue, 19 Nov 2019 20:29:21 GMT, Tue, 19 Nov 2019 20:29:21 GMT
       server: ViPR/1.0
       strict-transport-security: max-age=31536000; includeSubdomains; preload
       x-amz-id-2: 
       x-amz-request-id: 0a582f0b:16e7424ab55:383e:176e
       x-content-type-options: nosniff
       x-frame-options: DENY

7. Describe the issue:
Unable to  run the following command
pgbackrest --stanza=rebsrhpg001 stanza-create --log-level-console=trace

I'm surprised by the delimiter in the error message (URI/query):   delimiter=0.000000

Thanks

Christian

**Comment by dwsteele** [comment]
Yes, that delimiter is really odd.  We hardcode that to `/` as you can see here: https://github.com/pgbackrest/pgbackrest/blob/release/2.19/src/storage/s3/storage.c#L459.  I have no idea how it is ending up as 0.000000.

Could you run with `--log-level-console=debug` so we can see if this is happening before or after the call to `storageS3Request()`?

**Comment by kikijolicoeur** [comment]
Thank you for your quick answer.
Below the logs : 
2019-11-20 14:34:30.431 P00   INFO: stanza-create command begin 2.19: --log-level-console=debug --pg1-path=/var/lib/postgresql/data/10/main --pg1-socket-path=/var/run/postgresql --repo1-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo1-path=/pitr/vvhnoupn7cak56zkqdvu --repo1-s3-bucket=srh --repo1-s3-endpoint=storage-eb4.cegedim.cloud --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=eu-west-3 --repo1-type=s3 --stanza=rebsrhpg001
2019-11-20 14:34:30.431 P00  DEBUG:     common/lock::lockAcquire: (lockPath: {"/tmp/pgbackrest"}, stanza: {"rebsrhpg001"}, lockType: 2, lockTimeout: 0, failOnNoLock: true)
2019-11-20 14:34:30.431 P00  DEBUG:     common/lock::lockAcquire: => true
2019-11-20 14:34:30.431 P00  DEBUG:     config/load::cfgLoad: => void
2019-11-20 14:34:30.431 P00  DEBUG:     command/stanza/create::cmdStanzaCreate: (void)
2019-11-20 14:34:30.431 P00  DEBUG:     command/control/common::lockStopTest: (void)
2019-11-20 14:34:30.431 P00  DEBUG:     storage/storage::storageExists: (this: {type: posix, path: {"/"}, write: false}, pathExp: {"/tmp/pgbackrest/rebsrhpg001.stop"}, param.timeout: 0)
2019-11-20 14:34:30.431 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.431 P00  DEBUG:     storage/storage::storageExists: (this: {type: posix, path: {"/"}, write: false}, pathExp: {"/tmp/pgbackrest/all.stop"}, param.timeout: 0)
2019-11-20 14:34:30.431 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.431 P00  DEBUG:     command/control/common::lockStopTest: => void
2019-11-20 14:34:30.431 P00  DEBUG:     storage/s3/storage::storageS3New: (path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false, pathExpressionFunction: (function *), bucket: {"srh"}, endPoint: {"storage-eb4.cegedim.cloud"}, region: {"eu-west-3"}, partSize: 5242880, host: null, port: 443, timeout: 60000, verifyPeer: true, caFile: null, caPath: null)
2019-11-20 14:34:30.431 P00  DEBUG:     common/io/http/cache::httpClientCacheNew: (host: {"srh.storage-eb4.cegedim.cloud"}, port: 443, timeout: 60000, verifyPeer: true, caFile: null, caPath: null)
2019-11-20 14:34:30.431 P00  DEBUG:     common/io/http/cache::httpClientCacheNew: => {HttpClientCache}
2019-11-20 14:34:30.431 P00  DEBUG:     storage/s3/storage::storageS3New: => {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}
2019-11-20 14:34:30.431 P00  DEBUG:     storage/s3/storage::storageS3New: (path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: true, pathExpressionFunction: (function *), bucket: {"srh"}, endPoint: {"storage-eb4.cegedim.cloud"}, region: {"eu-west-3"}, partSize: 5242880, host: null, port: 443, timeout: 60000, verifyPeer: true, caFile: null, caPath: null)
2019-11-20 14:34:30.431 P00  DEBUG:     common/io/http/cache::httpClientCacheNew: (host: {"srh.storage-eb4.cegedim.cloud"}, port: 443, timeout: 60000, verifyPeer: true, caFile: null, caPath: null)
2019-11-20 14:34:30.431 P00  DEBUG:     common/io/http/cache::httpClientCacheNew: => {HttpClientCache}
2019-11-20 14:34:30.431 P00  DEBUG:     storage/s3/storage::storageS3New: => {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: true}
2019-11-20 14:34:30.431 P00  DEBUG:     db/helper::dbGet: (primaryOnly: false, primaryRequired: true)
2019-11-20 14:34:30.431 P00  DEBUG:     db/helper::dbGetId: (pgId: 1)
2019-11-20 14:34:30.431 P00  DEBUG:     protocol/helper::pgIsLocal: (hostId: 1)
2019-11-20 14:34:30.431 P00  DEBUG:     protocol/helper::pgIsLocal: => true
2019-11-20 14:34:30.431 P00  DEBUG:     postgres/client::pgClientNew: (host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout: 1800000)
2019-11-20 14:34:30.431 P00  DEBUG:     postgres/client::pgClientNew: => {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}
2019-11-20 14:34:30.431 P00  DEBUG:     db/db::dbNew: (client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null, applicationName: {"pgBackRest [stanza-create]"})
2019-11-20 14:34:30.431 P00  DEBUG:     db/db::dbNew: => {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}
2019-11-20 14:34:30.431 P00  DEBUG:     db/helper::dbGetId: => {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}
2019-11-20 14:34:30.431 P00  DEBUG:     db/db::dbOpen: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null})
2019-11-20 14:34:30.431 P00  DEBUG:     postgres/client::pgClientOpen: (this: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000})
2019-11-20 14:34:30.435 P00  DEBUG:     postgres/client::pgClientOpen: => {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}
2019-11-20 14:34:30.435 P00  DEBUG:     db/db::dbExec: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, command: {"set search_path = 'pg_catalog'"})
2019-11-20 14:34:30.435 P00  DEBUG:     db/db::dbQuery: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"set search_path = 'pg_catalog'"})
2019-11-20 14:34:30.435 P00  DEBUG:     postgres/client::pgClientQuery: (this: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, query: {"set search_path = 'pg_catalog'"})
2019-11-20 14:34:30.536 P00  DEBUG:     postgres/client::pgClientQuery: => null
2019-11-20 14:34:30.536 P00  DEBUG:     db/db::dbQuery: => null
2019-11-20 14:34:30.536 P00  DEBUG:     db/db::dbExec: => void
2019-11-20 14:34:30.536 P00  DEBUG:     db/db::dbQueryRow: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"select (select setting from pg_catalog.pg_settings where name = 'server_version_num')::int4, (select setting from pg_catalog.pg_settings where name = 'data_directory')::text, (select setting from pg_catalog.pg_settings where name = 'archive_mode')::text, (select setting from pg_catalog.pg_settings where name = 'archive_command')::text"})
2019-11-20 14:34:30.536 P00  DEBUG:     db/db::dbQuery: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"select (select setting from pg_catalog.pg_settings where name = 'server_version_num')::int4, (select setting from pg_catalog.pg_settings where name = 'data_directory')::text, (select setting from pg_catalog.pg_settings where name = 'archive_mode')::text, (select setting from pg_catalog.pg_settings where name = 'archive_command')::text"})
2019-11-20 14:34:30.536 P00  DEBUG:     postgres/client::pgClientQuery: (this: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, query: {"select (select setting from pg_catalog.pg_settings where name = 'server_version_num')::int4, (select setting from pg_catalog.pg_settings where name = 'data_directory')::text, (select setting from pg_catalog.pg_settings where name = 'archive_mode')::text, (select setting from pg_catalog.pg_settings where name = 'archive_command')::text"})
2019-11-20 14:34:30.636 P00  DEBUG:     postgres/client::pgClientQuery: => {VariantList}
2019-11-20 14:34:30.636 P00  DEBUG:     db/db::dbQuery: => {VariantList}
2019-11-20 14:34:30.636 P00  DEBUG:     db/db::dbQueryRow: => {VariantList}
2019-11-20 14:34:30.636 P00  DEBUG:     db/db::dbExec: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, command: {"set application_name = 'pgBackRest [stanza-create]'"})
2019-11-20 14:34:30.636 P00  DEBUG:     db/db::dbQuery: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"set application_name = 'pgBackRest [stanza-create]'"})
2019-11-20 14:34:30.636 P00  DEBUG:     postgres/client::pgClientQuery: (this: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, query: {"set application_name = 'pgBackRest [stanza-create]'"})
2019-11-20 14:34:30.736 P00  DEBUG:     postgres/client::pgClientQuery: => null
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbQuery: => null
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbExec: => void
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbOpen: => void
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbIsStandby: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null})
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbQueryColumn: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"select pg_catalog.pg_is_in_recovery()"})
2019-11-20 14:34:30.736 P00  DEBUG:     db/db::dbQuery: (this: {client: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, remoteClient: null}, query: {"select pg_catalog.pg_is_in_recovery()"})
2019-11-20 14:34:30.736 P00  DEBUG:     postgres/client::pgClientQuery: (this: {host: {"/var/run/postgresql"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, query: {"select pg_catalog.pg_is_in_recovery()"})
2019-11-20 14:34:30.837 P00  DEBUG:     postgres/client::pgClientQuery: => {VariantList}
2019-11-20 14:34:30.837 P00  DEBUG:     db/db::dbQuery: => {VariantList}
2019-11-20 14:34:30.837 P00  DEBUG:     db/db::dbQueryColumn: => {false}
2019-11-20 14:34:30.837 P00  DEBUG:     db/db::dbIsStandby: => false
2019-11-20 14:34:30.837 P00  DEBUG:     db/helper::dbGet: => {DbGetResult}
2019-11-20 14:34:30.837 P00  DEBUG:     protocol/helper::pgIsLocal: (hostId: 1)
2019-11-20 14:34:30.837 P00  DEBUG:     protocol/helper::pgIsLocal: => true
2019-11-20 14:34:30.837 P00  DEBUG:     storage/posix/storage::storagePosixNew: (path: {"/var/lib/postgresql/data/10/main"}, modeFile: 0640, modePath: 0750, write: false, pathExpressionFunction: null)
2019-11-20 14:34:30.837 P00  DEBUG:     storage/posix/storage::storagePosixNewInternal: (type: {"posix"}, path: {"/var/lib/postgresql/data/10/main"}, modeFile: 0640, modePath: 0750, write: false, pathExpressionFunction: null, pathSync: true)
2019-11-20 14:34:30.837 P00  DEBUG:     storage/posix/storage::storagePosixNewInternal: => {type: posix, path: {"/var/lib/postgresql/data/10/main"}, write: false}
2019-11-20 14:34:30.837 P00  DEBUG:     storage/posix/storage::storagePosixNew: => {type: posix, path: {"/var/lib/postgresql/data/10/main"}, write: false}
2019-11-20 14:34:30.837 P00  DEBUG:     postgres/interface::pgControlFromFile: (storage: {type: posix, path: {"/var/lib/postgresql/data/10/main"}, write: false})
2019-11-20 14:34:30.837 P00  DEBUG:     storage/storage::storageNewRead: (this: {type: posix, path: {"/var/lib/postgresql/data/10/main"}, write: false}, fileExp: {"global/pg_control"}, param.ignoreMissing: false, param.compressible: false)
2019-11-20 14:34:30.837 P00  DEBUG:     storage/storage::storageNewRead: => {type: posix, name: {"/var/lib/postgresql/data/10/main/global/pg_control"}, ignoreMissing: false}
2019-11-20 14:34:30.837 P00  DEBUG:     storage/storage::storageGet: (file: {type: posix, name: {"/var/lib/postgresql/data/10/main/global/pg_control"}, ignoreMissing: false}, param.exactSize: 512)
2019-11-20 14:34:30.837 P00  DEBUG:     storage/storage::storageGet: => {used: 512, size: 512, limit: <off>}
2019-11-20 14:34:30.837 P00  DEBUG:     postgres/interface::pgControlFromFile: => {version: 100000, systemId: 6652205002901602514, walSegmentSize: 16777216, pageChecksum: false}
2019-11-20 14:34:30.837 P00  DEBUG:     storage/storage::storageExists: (this: {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}, pathExp: {"<REPO:ARCHIVE>/archive.info"}, param.timeout: 0)
2019-11-20 14:34:30.837 P00  DEBUG:     storage/s3/storage::storageS3Exists: (this: {StorageS3}, file: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info"})
2019-11-20 14:34:30.837 P00  DEBUG:     storage/s3/storage::storageS3Request: (this: {StorageS3}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info"}, query: null, body: null, returnContent: true, allowMissing: true)
2019-11-20 14:34:30.839 P00  DEBUG:     common/io/http/client::httpClientRequest: (this: {HttpClient}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info"}, query: null, requestHeader: {authorization: <redacted>, content-length: '0', host: 'srh.storage-eb4.cegedim.cloud', x-amz-content-sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', x-amz-date: '20191120T133430Z'}, body: null, returnContent: true)
2019-11-20 14:34:30.850 P00  DEBUG:     common/io/http/client::httpClientRequest: => null
2019-11-20 14:34:30.850 P00  DEBUG:     storage/s3/storage::storageS3Request: => {StorageS3RequestResult}
2019-11-20 14:34:30.850 P00  DEBUG:     storage/s3/storage::storageS3Exists: => false
2019-11-20 14:34:30.850 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.850 P00  DEBUG:     storage/storage::storageExists: (this: {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}, pathExp: {"<REPO:ARCHIVE>/archive.info.copy"}, param.timeout: 0)
2019-11-20 14:34:30.850 P00  DEBUG:     storage/s3/storage::storageS3Exists: (this: {StorageS3}, file: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info.copy"})
2019-11-20 14:34:30.850 P00  DEBUG:     storage/s3/storage::storageS3Request: (this: {StorageS3}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info.copy"}, query: null, body: null, returnContent: true, allowMissing: true)
2019-11-20 14:34:30.850 P00  DEBUG:     common/io/http/client::httpClientRequest: (this: {HttpClient}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/archive.info.copy"}, query: null, requestHeader: {authorization: <redacted>, content-length: '0', host: 'srh.storage-eb4.cegedim.cloud', x-amz-content-sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', x-amz-date: '20191120T133430Z'}, body: null, returnContent: true)
2019-11-20 14:34:30.854 P00  DEBUG:     common/io/http/client::httpClientRequest: => null
2019-11-20 14:34:30.854 P00  DEBUG:     storage/s3/storage::storageS3Request: => {StorageS3RequestResult}
2019-11-20 14:34:30.854 P00  DEBUG:     storage/s3/storage::storageS3Exists: => false
2019-11-20 14:34:30.854 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.854 P00  DEBUG:     storage/storage::storageExists: (this: {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}, pathExp: {"<REPO:BACKUP>/backup.info"}, param.timeout: 0)
2019-11-20 14:34:30.854 P00  DEBUG:     storage/s3/storage::storageS3Exists: (this: {StorageS3}, file: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info"})
2019-11-20 14:34:30.854 P00  DEBUG:     storage/s3/storage::storageS3Request: (this: {StorageS3}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info"}, query: null, body: null, returnContent: true, allowMissing: true)
2019-11-20 14:34:30.854 P00  DEBUG:     common/io/http/client::httpClientRequest: (this: {HttpClient}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info"}, query: null, requestHeader: {authorization: <redacted>, content-length: '0', host: 'srh.storage-eb4.cegedim.cloud', x-amz-content-sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', x-amz-date: '20191120T133430Z'}, body: null, returnContent: true)
2019-11-20 14:34:30.859 P00  DEBUG:     common/io/http/client::httpClientRequest: => null
2019-11-20 14:34:30.859 P00  DEBUG:     storage/s3/storage::storageS3Request: => {StorageS3RequestResult}
2019-11-20 14:34:30.859 P00  DEBUG:     storage/s3/storage::storageS3Exists: => false
2019-11-20 14:34:30.859 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.859 P00  DEBUG:     storage/storage::storageExists: (this: {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}, pathExp: {"<REPO:BACKUP>/backup.info.copy"}, param.timeout: 0)
2019-11-20 14:34:30.859 P00  DEBUG:     storage/s3/storage::storageS3Exists: (this: {StorageS3}, file: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info.copy"})
2019-11-20 14:34:30.859 P00  DEBUG:     storage/s3/storage::storageS3Request: (this: {StorageS3}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info.copy"}, query: null, body: null, returnContent: true, allowMissing: true)
2019-11-20 14:34:30.859 P00  DEBUG:     common/io/http/client::httpClientRequest: (this: {HttpClient}, verb: {"HEAD"}, uri: {"/pitr/vvhnoupn7cak56zkqdvu/backup/rebsrhpg001/backup.info.copy"}, query: null, requestHeader: {authorization: <redacted>, content-length: '0', host: 'srh.storage-eb4.cegedim.cloud', x-amz-content-sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', x-amz-date: '20191120T133430Z'}, body: null, returnContent: true)
2019-11-20 14:34:30.863 P00  DEBUG:     common/io/http/client::httpClientRequest: => null
2019-11-20 14:34:30.863 P00  DEBUG:     storage/s3/storage::storageS3Request: => {StorageS3RequestResult}
2019-11-20 14:34:30.863 P00  DEBUG:     storage/s3/storage::storageS3Exists: => false
2019-11-20 14:34:30.863 P00  DEBUG:     storage/storage::storageExists: => false
2019-11-20 14:34:30.863 P00  DEBUG:     storage/storage::storageList: (this: {type: s3, path: {"/pitr/vvhnoupn7cak56zkqdvu"}, write: false}, pathExp: {"<REPO:ARCHIVE>"}, param.errorOnMissing: false, param.nullOnMissing: false, param.expression: null)
2019-11-20 14:34:30.863 P00  DEBUG:     storage/s3/storage::storageS3List: (this: {StorageS3}, path: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001"}, expression: null)
2019-11-20 14:34:30.863 P00  DEBUG:     storage/s3/storage::storageS3ListInternal: (this: {StorageS3}, path: {"/pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001"}, expression: null, recurse: false, callback: (function *), callbackData: *void)
2019-11-20 14:34:30.863 P00  DEBUG:     storage/s3/storage::storageS3Request: (this: {StorageS3}, verb: {"GET"}, uri: {"/"}, query: {delimiter: '/', list-type: '2', prefix: 'pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/'}, body: null, returnContent: true, allowMissing: false)
2019-11-20 14:34:30.863 P00  DEBUG:     common/io/http/client::httpClientRequest: (this: {HttpClient}, verb: {"GET"}, uri: {"/"}, query: {delimiter: '/', list-type: '2', prefix: 'pitr/vvhnoupn7cak56zkqdvu/archive/rebsrhpg001/'}, requestHeader: {authorization: <redacted>, content-length: '0', host: 'srh.storage-eb4.cegedim.cloud', x-amz-content-sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', x-amz-date: '20191120T133430Z'}, body: null, returnContent: true)
2019-11-20 14:34:30.923 P00  DEBUG:     common/io/http/client::httpClientRequest: => null
2019-11-20 14:34:30.923 P00  DEBUG:     common/exit::exitSafe: (result: 0, error: true, signalType: 0)
2019-11-20 14:34:30.924 P00 DETAIL: tls statistics: objects 1, sessions 1, requests 5, retries 0
2019-11-20 14:34:30.924 P00   INFO: http statistics: objects 1, sessions 1, requests 5, retries 0, closes 1
2019-11-20 14:34:30.924 P00   INFO: stanza-create command end: aborted with exception [039]
2019-11-20 14:34:30.924 P00  DEBUG:     common/lock::lockRelease: (failOnNoLock: false)
2019-11-20 14:34:30.924 P00  DEBUG:     common/lock::lockRelease: => true
2019-11-20 14:34:30.924 P00  DEBUG:     common/exit::exitSafe: => 39
2019-11-20 14:34:30.924 P00  DEBUG:     main::main: => 39

**Comment by dwsteele** [comment]
Well, that all looks normal.  Can you run with `--log-level-console=trace` and send everything after the last `storage/s3/storage::storageS3Request: (` line?

*Note that trace can expose secrets (just the key, not the secret key as far as we know) so you'll want scrub the log.*

**Comment by kikijolicoeur** [comment]
Hello. Thanks for your investigations. 
I'm so sorry that you lost your time. With pgbackrest logs I discovered the root cause of my problem.
Some explanations:
There are two modes to adressing bucket : virtual host style et path style (see [https://aws.amazon.com/fr/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/](url)) for details. I saw in pgbackrest logs that pgbackrest used the virtual host style. So if your endpoint is myendpoint.com and your bucket is mybucket, pgbackrest connects to https://mybucket.myendpoint.com. With DELL ECS S3, there is a concept of namespace. A S3 user in DELL ECS belongs to a namespace. To access on a bucket with virtual host style in DELL ECS we must add the namespace in the endpoint. So, if my namespace names mynamespace, the correct URL is https://mybucket.mynamespace.myendpoint.com.  So I had the namespace in repo1-s3-endpoint and a DNS alias and all works fine. 
Question : is it possible to use path style in pgbackrest ?
Thanks for your work.

**Comment by dwsteele** [comment]
> I'm so sorry that you lost your time. With pgbackrest logs I discovered the root cause of my problem.

No worries -- it looked like a bug to me, too.

I did figure out where `0.00000` was coming from, though.  It's from the %XXf encoding in the URL being interpreted as format specifiers somewhere.  Not sure if that is us or your env but we'll look into it.

> So I had the namespace in repo1-s3-endpoint and a DNS alias and all works fine.

Makes sense.

> Question : is it possible to use path style in pgbackrest ?

No.  We intended to add it but then Amazon deprecated the feature so it didn't seem worth the trouble.  It might still happen, but we currently have no plans to do it.
